drm-howto/modeset-atomic.c

```diff
@@ -745,18 +790,31 @@ static int modeset_atomic_modeset(int fd, struct modeset_dev *dev)
 
 	req = drmModeAtomicAlloc();
 
-	if (set_connector_property(req, &dev->connector, dev->conn_id, "CRTC_ID", dev->crtc_id) < 0)
+	/* prepare our connector object to set the CRTC id that
+	 * the connector is using */
+	if (set_connector_property(req, &dev->connector, dev->conn_id, 
+							   "CRTC_ID", dev->crtc_id) < 0)
 		return -1;
 
+	/* Creates a blob property that receives dev->mode value. Then, the CRTC
+	 * object property "MODE_ID" points to the id of the blob. This usually
+	 * happens for properties that are not simple types. In this case, dev->mode
+	 * is a struct. But we could have a property that expects the id of a blob
+	 * that holds an array, for instance.
+	 */
 	if (drmModeCreatePropertyBlob(fd, &dev->mode, sizeof(dev->mode), &blob_id) != 0)
 		return -1;
-
 	if (set_crtc_property(req, &dev->crtc, dev->crtc_id, "MODE_ID", blob_id) < 0)
 		return -1;
 
+	/* prepare our CRTC object to be set as active */
 	if (set_crtc_property(req, &dev->crtc, dev->crtc_id, "ACTIVE", 1) < 0)
 		return -1;
 
+	/* Set the connector and CRTC properties in a single IOCTL. Here we leave it
+	 * as a blocking call, since we can't start to use the KMS API before
+	 * setting everything up.
```

Comment:
I'm not sure what "since we can't start to use the KMS API before setting everything up." refers to. This already is using KMS API, just in a blocking way. Do you mean pageflipping instead?

Reply:
> Do you mean pageflipping instead?

Yes, sorry about that!